python/cugraph/structure/shuffle.py

```diff
@@ -0,0 +1,46 @@
+import math
+from dask.dataframe.shuffle import rearrange_by_column
+
+
+def get_n_workers():
+    from dask.distributed import default_client
+    client = default_client()
+    return len(client.scheduler_info()['workers'])
+
+
+def get_2D_div(ngpus):
+    pcols = int(math.sqrt(ngpus))
+    while ngpus % pcols != 0:
+        pcols = pcols - 1
+    return int(ngpus/pcols), pcols
+
+
+def _set_partitions_pre(df, num_verts, prows, pcols):
+    rows_per_div = math.ceil(num_verts/prows)
+    cols_per_div = math.ceil(num_verts/pcols)
+    partitions = df['src'].floordiv(rows_per_div) * pcols + df['dst'].floordiv(cols_per_div)
```

Comment:
Can we do this? If I am not mistaken, we apply a hash function to edge source & destination vertices to determine the partition each edge belongs to. So, each partition will have roughly same number of rows or columns but they will not be same (and can't be  determined in this way). Am I missing something?

Reply:
updated